File: src/Server/Infrastructure/Persistence/Entities/Notification.cs
using Server.Core.Common;
using Server.Infrastructure.Identity;
using Server.Infrastructure.Persistence.Entities.Enums;

namespace Server.Infrastructure.Persistence.Entities;

public class Notification : BaseEntity
{
    public Guid UserId { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Message { get; set; } = string.Empty;

    public NotificationType Type { get; set; }

    public NotificationPriority Priority { get; set; } = NotificationPriority.Normal;

    public string? RelatedEntityType { get; set; }

    public Guid? RelatedEntityId { get; set; }

    public bool IsRead { get; set; }

    public DateTime? ReadAt { get; set; }

    public string? ActionUrl { get; set; }

    public ApplicationUser? User { get; set; }
}

File: src/client/src/types/notification.ts
export enum NotificationType {
  AppointmentReminder = 0,
  AppointmentCancelled = 1,
  AppointmentCompleted = 2,
  InvoiceIssued = 3,
  InvoiceOverdue = 4,
  PaymentReceived = 5,
  LabResultReady = 6,
  PrescriptionReady = 7,
  FollowUpDue = 8,
  SystemAlert = 9,
}

export enum NotificationPriority {
  Low = 0,
  Normal = 1,
  High = 2,
  Urgent = 3,
}

export interface NotificationItem {
  id: string;
  title: string;
  message: string;
  type: NotificationType;
  priority: NotificationPriority;
  isRead: boolean;
  createdAt: string;
  actionUrl?: string | null;
}

export interface NotificationDetail extends NotificationItem {
  readAt?: string | null;
  actionUrl?: string | null;
  relatedEntityType?: string | null;
  relatedEntityId?: string | null;
}

export interface UnreadCount {
  count: number;
}

export interface MarkAsReadRequest {
  notificationIds: string[];
}
